libavfilter/vf_overlay_videotoolbox.m

```diff
@@ -392,56 +410,34 @@ static av_cold int overlay_videotoolbox_init(AVFilterContext *ctx)
     }
 }
 
-static int do_config_input(AVFilterLink *inlink) API_AVAILABLE(macos(10.11), ios(9.0))
+static int do_config_output(AVFilterLink *link) API_AVAILABLE(macos(10.11), ios(9.0))
 {
-    AVFilterContext *avctx = inlink->dst;
+    AVFilterContext *avctx = link->src;
+    AVFilterLink *inlink_main = avctx->inputs[0];
+    AVFilterLink *inlink_overlay = avctx->inputs[1];
     OverlayVideoToolboxContext *ctx = avctx->priv;
+    AVHWFramesContext *main_frames, *output_frames;
     AVBufferRef *input_ref;
-    AVHWFramesContext *input_frames;
+    int ret = 0;
 
-    if (!inlink->hw_frames_ctx) {
-        av_log(avctx, AV_LOG_ERROR, "A hardware frames reference is "
+    if (!inlink_main->hw_frames_ctx ||
+        !inlink_overlay->hw_frames_ctx) {
```

Comment:
Not sure if Metal, CI and CV handle main and overlay correctly when they have different videotoolbox devices or HW contexts.

Reply:
If not, `av_assert0(main_frames->device_ctx == overlay_frames->device_ctx)`.

Reply:
Actually "different devices" concept does not exist in VideoToolbox:

https://github.com/gnattu/jellyfin-ffmpeg/blob/8dd6758d45176725ae882896d08c998df94d11fb/libavutil/hwcontext_videotoolbox.c#L751-L760

The native API provided by Apple only have the concept of "Session", and the device running the session is not user-controllable.